### PR TITLE
regenerate client code with latest k8s api

### PR DESCRIPTION
**Pull request by eastlondoner**
So that it is compatible with python 3.7

### CODE REVIEW

**Comment by roycaihw** [comment]
More thoughts on this: 

Kubernetes supports three minor releases at the same time. Currently it's 1.10, 1.11 and 1.12. This python client repo follows similar policy and we currently support v6, v7 and v8. Kubernetes v1.13.0 is planned to be released in less a month ([Dec. 3rd](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.13)). After k8s 1.13 is released, 1.10 will not be supported, and we will start releasing python client v9 and dropping v6.

At this point, given the short period of supporting 1.10, I'd suggest switching to more recent versions of Kubernetes, instead of releasing a python client v6.0.1 to support Python 3.7

**Comment by eastlondoner** [comment]
@roycaihw 
>The python 3.7 support issue was about the reserved keyword workaround. 
Yes. In order to get the fix for the reserved keyword it is necessary to do what I have done in this PR - do you disagree?

> We already have two GA releases that support Python 3.7 (v7.0.0 and v8.0.0).
I would like to use python 3.7 with v6 because I want to use K8 1.10 which is supposed to be supported now.

> At this point, given the short period of supporting 1.10, I'd suggest switching to more recent versions of Kubernetes, instead of releasing a python client v6.0.1 to support Python 3.7

I suggest that you merge my PR and support all the versions of K8s that are supposed to be supported.

That sounds good in principle. We are using hosted Kubernetes with GCP and 1.11 has only been available for general use for less than a week, we need some time to ensure that we can move to 1.11 without problems.

Do you have an actual reason not to merge this PR?

File: kubernetes/README.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ Python 2.7 and 3.4+
 If the python package is hosted on Github, you can install directly from Github
 
 ```sh
-pip install git+https://github.com/kubernetes-client/python.git
+pip install git+https://github.com/kubernetes-kubernetes.client/python.git
 ```
-(you may need to run `pip` with root permission: `sudo pip install git+https://github.com/kubernetes-client/python.git`)
+(you may need to run `pip` with root permission: `sudo pip install git+https://github.com/kubernetes-kubernetes.client/python.git`)
```
Comment (roycaihw) [inline kubernetes/README.md]: Please manually restore the commands in L20 and L22. I believe we have some misconfig of automation/swagger-codegen that writes the extra `kubernetes.`

File: kubernetes/swagger.json.unprocessed [hunk 2]
```diff
@@ -79877,7 +79877,7 @@
           "type": "string"
         },
         "mountPropagation": {
-          "description": "mountPropagation determines how mounts are propagated from the host to container and the other way around. When not set, MountPropagationHostToContainer is used. This field is beta in 1.10.",
+          "description": "mountPropagation determines how mounts are propagated from the host to container and the other way around. When not set, MountPropagationNone is used. This field is beta in 1.10.",
```
Comment (roycaihw) [inline kubernetes/swagger.json.unprocessed]: Looks like there is no significant api change from openapi spec :)

**Comment by eastlondoner** [comment]
Thanks @roycaihw.
I have updated the README, CHANGELOG, setup.py and tox.ini

File: tox.ini [hunk 3]
```diff
@@ -34,6 +34,11 @@ commands =
    python -V
    {toxinidir}/scripts/kube-init.sh py.test -vvv -s []
 
+[testenv:py37-functional]
+commands =
+   python -V
+   {toxinidir}/scripts/kube-init.sh py.test -vvv -s []
+
```
[the remark this answers is not in the document]
Comment (micw523) [reply tox.ini]: See build #1069 https://travis-ci.org/kubernetes-client/python/builds/452318635?utm_source=github_status&utm_medium=notification

[the remark this answers is not in the document]
Comment (eastlondoner) [reply tox.ini]: ah, well I guess this is future proof at least

**Comment by roycaihw** [comment]
I realized that I missed mentioning the step to update version constant. Please regenerate the client with the following steps:

1. update `CLIENT_VERSION` in `scripts/constants.py` to be 6.1.0 (ref: https://github.com/kubernetes-client/python/blob/master/devel/release.md#update-release-tags)
2. run `scripts/update-client.sh` and remove the extra "kubernetes." in README (swagger-codegen will update the client versions for you, so you don't have to manually change them)
3. add release note to CHANGELOG

File: tox.ini (hunk 3 above)
[the remark this answers is not in the document]
Comment (roycaihw) [reply tox.ini]: We haven't supported Python 3.7 in our travis-ci yet. Please see https://github.com/kubernetes-client/python/issues/584

Changes (`tox.ini` and `.travis.yml`) need to be checked into master branch in both `python` and `python-base`. I'm more comfortable with having separate PRs for those changes and this PR including necessary client re-generation. WDYT?

[the remark this answers is not in the document]
Comment (micw523) [reply tox.ini]: Me too. I prefer to wait to see Travis CI support Python 3.7 and make sure the keywords are correct, and then check them into the master branch.

**Comment by eastlondoner** [comment]
Done.
I reverted and started over. Changes are in a single commit. No changes to tox.ini

**Comment by eastlondoner** [comment]
Thanks :-D this helps us to upgrade our stack smoothly as we work towards 1.11

**Pull request merged**